### PR TITLE
Detail explain how to install build dependencies on apt based system

### LINKED ISSUE #32: Document how to make `apt-get build-dep python3.5` to work
I.e. if you get the error about needing to update `sources.list`, then point out how to do that.

### PR DESCRIPTION
**Pull request by zhsj**
Closes #32

### CODE REVIEW

File: setup.rst [hunk 1]
```diff
@@ -125,7 +125,22 @@ On **Fedora** and other ``DNF`` based systems::
    $ sudo dnf builddep python3
 
 On **Debian**, **Ubuntu**, and other ``apt`` based systems, try to get the
-dependencies for the Python version that you're working on e.g.::
+dependencies for the Python using the ``apt``.
+
+First please make sure you have enabled the source packages in source list.
+You can do this by adding the following line to ``/etc/apt/sources.list``.
```
Comment (willingc) [inline setup.rst]: ~~following line~ the location of the source packages, including URL and distribution name,

File: setup.rst [hunk 2]
```diff
@@ -125,7 +125,22 @@ On **Fedora** and other ``DNF`` based systems::
...
+Take Ubuntu Xenial for example::
+
+   deb-src http://archive.ubuntu.com/ubuntu/ xenial main
+
+Other systems like Debian should change this to the corresponding URL and
```
Comment (willingc) [inline setup.rst]: Perhaps change to:

For other distributions, like Debian, change the URL and name to correspond with the specific distribution.

File: setup.rst [hunk 3]
```diff
@@ -125,7 +125,22 @@ On **Fedora** and other ``DNF`` based systems::
...
+distribution name.
+
+Then you should update the packages index::
+
+   $ sudo apt-get update
+
+Now you can install the dependencies via ``apt``::
```
Comment (willingc) [inline setup.rst]: ... the build dependencies ...

File: setup.rst (hunk 1 above)
Comment (zhsj) [inline setup.rst]: thanks for the suggestion, I modify it to URL, distribution name(xenial) and component name(main)

**Comment by willingc** [comment]
@Mariatta Will you give this a quick review and merge?

File: setup.rst [hunk 4]
```diff
@@ -125,7 +125,23 @@ On **Fedora** and other ``DNF`` based systems::
    $ sudo dnf builddep python3
 
 On **Debian**, **Ubuntu**, and other ``apt`` based systems, try to get the
-dependencies for the Python version that you're working on e.g.::
+dependencies for the Python using the ``apt``.
```
Comment (DimitrisJim) [inline setup.rst]: You can drop "the" before "Python" here

Comment (berkerpeksag) [inline setup.rst]: I'm not a native speaker, but the second part of the new sentence reads awkward:

```
[...] try to get the dependencies for the Python using the ``apt``.
```
How about something like "[...] you can get the dependencies for the Python version that you're working on by using the \`\`apt\`\` command"?

File: setup.rst [hunk 5]
```diff
@@ -125,7 +125,23 @@ On **Fedora** and other ``DNF`` based systems::
...
+
+First please make sure you have enabled the source packages in source list.
```
Comment (DimitrisJim) [inline setup.rst]: I think removing "please" would be better, i.e: "First, make sure ..."
Additionally,  "source" -> "sources".

File: setup.rst (hunk 4 above)
Comment (DimitrisJim) [inline setup.rst]: That definitely reads better, +1 for it.

Comment (zhsj) [inline setup.rst]: Reads better indeed, thanks

**Comment by Mariatta** [comment]
Thanks @zhsj and congrats for your first contribution to the Dev Guide :tada:

**Pull request merged**